docs/docs/ref-03-component-specs.md

```diff
@@ -148,6 +148,8 @@ componentWillReceiveProps: function(nextProps) {
 
 > Note:
 >
+> It is **incorrect** for code executed during this lifecycle method to assume that props have changed.  To understand why, read [A implies B does not imply B implies A](/react/blog/2016/01/08/A-implies-B-does-not-imply-B-implies-A.html)
```

Comment:
I want to make this point about it being incorrect abundantly clear, since people seem to make this assumption even after they are explicitly told that there is no guarantee that props have changed.

BTW, all sentences in the entire site should be interpreted logically; people should be careful in their parsing of all documentation.  We work very hard to ensure that the documentation is as precise as possible, often spending hours nitpicking the choice of only a few words.  I'll deemphasize the incorrectness to avoid hurting people's feelings, but I do want to call out the fact that it's a common mistake.